### PR TITLE
Add AI Generator Editor

### PR DESCRIPTION
**Pull request by yasha-dev1**
**Changes proposed in this Pull Request**
Add AI Wizard Editor

**Checklist**
- [x] My code follows the style guidelines of this project
- [x] My commits follow the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

Close QualityUnit/web-issues#[ISSUE_NUMBER]

### CODE REVIEW

**Comment by yasha-dev1** [comment]
I know [ContentGeneratorPanel](https://github.com/QualityUnit/wp-urlslab/pull/988/files#diff-bb5793026cab88841a37d2222b000e6035af3987f6de727706fe64f4f47ce268) is a mess in terms of `useState`. In the next PR we will use the global store to left the states up, cause we need to access it outside the component

**Comment by ihdk** [comment]
@yasha-dev1 is there any issue for this? just to know more what is going on and what was implemented...

**Comment by yasha-dev1** [comment]
> @yasha-dev1 is there any issue for this? just to know more what is going on and what was implemented...

Not Exactly, cuz its related to https://github.com/QualityUnit/web-issues/issues/1756
basically the functionality of the popup (without the input text in case of preselection). we should be able to reuse the same flow/component in the popup

File: admin/src/components/generator/ContentGeneratorPanel.jsx
Comment (ihdk) [inline admin/src/components/generator/ContentGeneratorPanel.jsx:1]: missing key attribute on react nodes in the loop:

File: admin/src/components/generator/ContentGeneratorPanel.jsx [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+								key={ postType }
+								items={ postTypes }
+								name="post_type_menu"
+								defaultAccept
+								autoClose
+								defaultValue={ postType }
+								onChange={ ( val ) => setPostType( val ) }
+							/>
+
+							<Button onClick={ handleCreatePost }>Create Post</Button>
```
Comment (ihdk) [inline admin/src/components/generator/ContentGeneratorPanel.jsx:468]: missing localization functions

**Review by ihdk — CHANGES_REQUESTED** [review]
I just left some comments for the ContentGeneratorPanel.

If states will be handled later, than ok, it's the probably reason why opening/closing of all dropdowns in generator tooks so long...

File: admin/src/components/generator/ContentGeneratorPanel.jsx (hunk 1 above)
Comment (ihdk) [reply admin/src/components/generator/ContentGeneratorPanel.jsx]: @yasha-dev1 buttons stucked on each other will stay as is for now?

Comment (yasha-dev1) [reply admin/src/components/generator/ContentGeneratorPanel.jsx]: Yes, after using the framework, we'll work on the UI

**Review by ihdk — APPROVED** [review]

File: admin/src/components/generator/ContentGeneratorPanel.jsx [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+
+	const contextTypePromptPlaceholder = {
+		NO_CONTEXT: __( 'Your prompt to be used for generating text…' ),
+		URL_CONTEXT: __( 'The prompt to be used for generating text from each url…' ),
+		DOMAIN_CONTEXT: __( 'The prompt to be used to generate text from relevant content in your whole domain…' ),
+		SERP_CONTEXT: __( 'The prompt to be used to generate text from top SERP Results targeting your keyword…' ),
+	};
+
+	useEffect( () => {
+		getPostTypes().then( async ( res ) => {
```
Comment (jremen) [inline admin/src/components/generator/ContentGeneratorPanel.jsx:69]: avoid using then, use await directly

File: admin/src/api/generatorApi.js
Comment (jremen) [inline admin/src/api/generatorApi.js:1]: We should cache this using the tanstack query, not just fetch it

File: admin/src/components/generator/ContentGeneratorPanel.jsx [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+	augmentWithURLContext, createPost,
+	getAugmentProcessResult, getPostTypes,
+} from '../../api/generatorApi';
+import { Editor as TinyMCE } from '@tinymce/tinymce-react/lib/cjs/main/ts/components/Editor';
+import { fetchLangs } from '../../api/fetchLangs';
+
+function ContentGeneratorPanel() {
+	const { __ } = useI18n();
+	const editorRef = useRef( null );
+	const [ editorVal, setEditorVal ] = useState( '' );
```
Comment (jremen) [inline admin/src/components/generator/ContentGeneratorPanel.jsx:24]: huge mess

File: admin/src/components/generator/ContentGeneratorPanel.jsx [hunk 4]
```diff
@@ -0,0 +1,482 @@
...
+
+	useEffect( () => {
+		getPostTypes().then( async ( res ) => {
+			const rsp = await res.json();
+			setPostTypes( rsp );
+		} );
+	}, [] );
+
+	const handleCreatePost = async () => {
+		createPost( editorVal, title, postType ).then( async ( res ) => {
```
Comment (jremen) [inline admin/src/components/generator/ContentGeneratorPanel.jsx:76]: avoid using then, use await directly

File: admin/src/components/generator/ContentGeneratorPanel.jsx [hunk 5]
```diff
@@ -0,0 +1,482 @@
...
+			const rsp = await res.json();
+			setGeneratedPostLink( decodeURIComponent( rsp.edit_post_link ) );
```
Comment (jremen) [inline admin/src/components/generator/ContentGeneratorPanel.jsx:78]: why not save results with tanstack query as separate query?

**Review by jremen — CHANGES_REQUESTED** [review]
rewrite using tanstack query so you can avoid all states writing.
This is a mess. Will not pass through Code review

**Comment by jremen** [comment]
@yasha-dev1 
Can't pass through code review. Write the results to cached queries. If you are not aware of this, please pass the issue to me or Ivan…

File: admin/src/components/generator/ContentGeneratorPanel.jsx (hunk 2 above)
Comment (yasha-dev1) [reply admin/src/components/generator/ContentGeneratorPanel.jsx]: fixed

File: admin/src/components/generator/ContentGeneratorPanel.jsx (hunk 3 above)
Comment (yasha-dev1) [reply admin/src/components/generator/ContentGeneratorPanel.jsx:25]: As discussed, we're gonna lift the state up and use global storage for that. so all `useState` will be removed from this component

File: admin/src/components/generator/ContentGeneratorPanel.jsx (hunk 4 above)
Comment (yasha-dev1) [reply admin/src/components/generator/ContentGeneratorPanel.jsx]: fixed

File: admin/src/components/generator/ContentGeneratorPanel.jsx (hunk 5 above)
Comment (yasha-dev1) [reply admin/src/components/generator/ContentGeneratorPanel.jsx]: This request is for creating a new post, it doesn't make sense to cache this

File: admin/src/api/generatorApi.js
Comment (yasha-dev1) [reply admin/src/api/generatorApi.js:1]: fixed

**Review by jremen — CHANGES_REQUESTED** [review]
At least move the logic to the separate hook and use object as a state and not separate every single item into separate useState. You can put any data type into state…

**Review by jremen — APPROVED** [review]

**Pull request merged**